### PR TITLE
fix for FTP urls

### LINKED ISSUE #142: `hash_url` broken for libmagic
when running `hash_url` on `ftp://ftp.astron.com/pub/file/file-5.33.tar.gz` I get an `'_io.BufferedReader' object has no attribute 'length'` error is that expected?

### PR DESCRIPTION
**Pull request by scopatz**
closes #142, CC @CJ-Wright

### CODE REVIEW

File: tests/test_tools.py [hunk 1]
```diff
@@ -37,5 +37,9 @@ def test_render_authors():
         assert render_authors(a) == b
 
 
-def test_hash_url():
+def test_hash_url_http():
     hash_url('http://python.org')
+
+
+def test_hash_url_ftp():
+    has_url('ftp://ftp.astron.com/pub/file/file-5.33.tar.g')
```
Comment (CJ-Wright) [inline tests/test_tools.py]: I think it is `hash_url`

File: tests/test_tools.py [hunk 2]
```diff
@@ -37,5 +37,9 @@ def test_render_authors():
         assert render_authors(a) == b
 
 
-def test_hash_url():
+def test_hash_url_http():
     hash_url('http://python.org')
+
+
+def test_hash_url_ftp():
+    hash_url('ftp://ftp.astron.com/pub/file/file-5.33.tar.g')
```
Comment (CJ-Wright) [inline tests/test_tools.py]: `file-5.33.tar.gz`

Comment (scopatz) [reply tests/test_tools.py]: can't sleep must zzzz

**Comment by CJ-Wright** [comment]
(this truly was the PR of spelling errors :smile:)

**Comment by scopatz** [comment]
hahah

**Pull request merged**